From 4beb9a405abbabfdfc9f1980b498a884208e4106 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Wed, 26 Aug 2015 21:23:21 +0530
Subject: [PATCH] wcove-gpio: Use schedule_work instead of system worker.

Due to delay in event processing with system worker,
vbus enable/disable is not meeting the  required timing.
Hence moved to  schedule_work.

Change-Id: Icd3009e1ab34e9aac76941fb927769de7e05b21a
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-17332
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
---
 drivers/misc/intel_wcove_gpio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/intel_wcove_gpio.c b/drivers/misc/intel_wcove_gpio.c
index d95f5354fa3f..5675bfd9899f 100644
--- a/drivers/misc/intel_wcove_gpio.c
+++ b/drivers/misc/intel_wcove_gpio.c
@@ -196,7 +196,7 @@ static int wcgpio_check_events(struct wcove_gpio_info *info,
 	list_add_tail(&evt->node, &info->gpio_queue);
 	spin_unlock(&info->gpio_queue_lock);
 
-	queue_work(system_nrt_wq, &info->gpio_work);
+	schedule_work(&info->gpio_work);
 	return 0;
 }
 
-- 
1.9.1

